lib/pry/config/behavior.rb

```diff
@@ -29,16 +33,19 @@ def assign(attributes, default = nil)
         end
 
         #
-        # Returns a new Behavior, recursively walking attributes.
+        # @example
+        #   c = Pry::Config.from_hash(foo: {bar: {baz: 42}})
+        #   c.foo.bar.class # => Pry::Config
+        #   c.foo.bar.baz   # => 42
         #
         # @param [Hash] attributes
-        #   a hash to initialize an instance of self with.
         #
-        # @param [Pry::Config, nil] default
-        #   a default, or nil for none.
+        # @param [Pry::Config::Behavior, nil] default
+        #   A default, or nil for none.
```

Comment:
Would be nice to explain params on the same line. See how YARD does it https://github.com/lsegal/yard/blob/master/lib/yard/docstring.rb#L33

This applies to all the tags. YARD uses one line, without splitting.

Reply:
i don't find that to be as readable.

```ruby
# @param [String] foo the foo string
# @param [Hash] options the options hash
# @return [String] some sentance describing a string
```
vs
```ruby
#
# @param [String] foo
#   The foo string
#
# @param [Hash] options
#   The options hash
# 
# @return [String]
#   Some sentence describing a string.
#
```

granted it takes up more lines but for me it is much more human parseable and readable.

Reply:
It's not about personal preference but about consistency. We use the compact style everywhere else in this project (unless the code is written by you) and to prove my point I linked to YARD (it's more common). While you can consume YARD here in the code it's meant to be read in your browser and prettifying it in the code takes a lot of space.

Reply:
You started changing the format that we use to document code in this project out of thin air and you should expect this kind of conversation. I've never seen this kind of format before so maybe you can link to places that inspired you. In my opinion this style is ugly and cumbersome. You can highlight YARD code in emacs https://github.com/pd/yard-mode.el. Maybe something like that exists for your editor, too.

That said, I don't think I'll convince you otherwise so this discussion is fruitless. I generated YARD docs based on this PR and they still look good.

Reply:
> You started changing the format that we use to document code in this project out of thin air and you should expect this kind of conversation.

I didn't change the format used by Pry; there never was a standard format or coding style used by Pry. every contributor introduced their own style, and style discussions never came up during review - that i ever saw at least.

this style nonsense has been introduced by you, and now i am told i am changing some style that always existed, but it didn't. 

i did not just arrive and start using this format from nowhere one day, i used this format for documentation since i started working on Pry and i believe my contributions predate yours, cirwins, and rfs.

> That said, I don't think I'll convince you otherwise so this discussion is fruitless.

you're right, i won't adopt the squashed format because it impedes on my ability to write good docs and that matters more than some silly style guide.

Reply:
as for the question, "where did i find this style for docs that i use?", from https://github.com/postmodern